Add PHP 8.4 to the list of allowed PHP versions

settermjd opened the pull request:
As PHP's been out for a couple of months now, it seems worth adding it to the list of allowed versions. I've not removed 8.1 as it still has security support until 31.12.2025.

settermjd committed:
Add PHP 8.4 to the list of allowed PHP versions

As PHP's been out for a couple of months now, it seems worth adding it
to the list of allowed versions. I've not removed 8.1 as it still has
security support until 31.12.2025.

asgrim commented on composer.json:
```diff
@@ -3,7 +3,7 @@
     "type": "library",
     "require": {
-        "php": "8.1.*|8.2.*|8.3.*",
+        "php": "8.1.*|8.2.*|8.3.*|8.4.*",
```
We could do with adding the new PHP versions to the CI matrix too, would you be able to add them there too please? We didn't adopt Laminas CI matrix on this yet; that would be a good addition in future though :100:

settermjd replied:
Oops. Sure. Sorry for not thinking about that. Will sort out both points.

settermjd replied:
Hey @asgrim. 👋🏽  If you're okay with the additional changes made, then could you merge the PR, pls?

settermjd committed:
Update the workflows to list all the required PHP versions

settermjd committed:
Update the package's dependencies to allow for upgrading to 8.4

This change updates Composer's lock file and Psalm to a minimum version
of 6.5 so that the package can use PHP 8.4. I hope that I've done this
the right way.

settermjd committed:
Remove unused Psalm issue type suppression configuration

This change removes the suppression of unused issue types so that Psalm
successfully completes.

asgrim commented:
Nice one, thanks @settermjd :tada:
